diff -Naur a/gtksourceview/gtksourcegutter.c b/gtksourceview/gtksourcegutter.c
--- a/gtksourceview/gtksourcegutter.c
+++ b/gtksourceview/gtksourcegutter.c
@@ -831,7 +831,6 @@
 	    (!gtk_text_buffer_get_selection_bounds (buffer, &cur, &sel) ||
              gtk_text_iter_get_line (&cur) == gtk_text_iter_get_line (&sel)))
 	{
-		GtkRoot *root;
 		GdkRGBA highlight;
 		guint cursor_line;
 
@@ -839,10 +838,7 @@
 
 		if (cursor_line >= gtk_source_gutter_lines_get_first (gutter->lines) &&
 		    cursor_line <= gtk_source_gutter_lines_get_last (gutter->lines) &&
-		    _gtk_source_view_get_current_line_number_background (gutter->view, &highlight) &&
-		    (root = gtk_widget_get_root (GTK_WIDGET (gutter->view))) &&
-		    GTK_IS_WINDOW (root) &&
-		    gtk_window_is_active (GTK_WINDOW (root)))
+		    _gtk_source_view_get_current_line_number_background (gutter->view, &highlight))
 		{
 			int width = gtk_widget_get_width (widget);
 			double height;
diff -Naur a/gtksourceview/gtksourceview.c b/gtksourceview/gtksourceview.c
--- a/gtksourceview/gtksourceview.c
+++ b/gtksourceview/gtksourceview.c
@@ -2909,8 +2909,6 @@
 
 	if (layer == GTK_TEXT_VIEW_LAYER_BELOW_TEXT)
 	{
-		GtkRoot *root;
-
 		/* Now draw the background pattern, which might draw above the
 		 * right-margin area for additional texture. We can't really optimize
 		 * these too much since they move every scroll. Otherwise we'd move
@@ -2927,10 +2925,7 @@
 		 */
 		if (gtk_widget_is_sensitive (GTK_WIDGET (view)) &&
 		    priv->highlight_current_line &&
-		    priv->current_line_background_color_set &&
-		    (root = gtk_widget_get_root (GTK_WIDGET (view))) &&
-		    GTK_IS_WINDOW (root) &&
-		    gtk_window_is_active (GTK_WINDOW (root)))
+		    priv->current_line_background_color_set)
 		{
 			gtk_source_view_paint_current_line_highlight (view, snapshot);
 		}
